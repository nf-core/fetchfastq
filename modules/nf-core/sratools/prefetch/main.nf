process SRATOOLS_PREFETCH {
    tag "$id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/sra-tools:3.0.8--h9f5acd7_0' :
        'biocontainers/sra-tools:3.0.8--h9f5acd7_0' }"

    input:
    Tuple2<Map,String> input
    Path ncbi_settings
    Path certificate
    String prefetch_args = ''
    String retry_args = '5 1 100'  // <num retries> <base delay in seconds> <max delay in seconds>

    output:
    Tuple2<Map,String> sra = input

    topic:
    [ task.process, 'sratools', eval("prefetch --version 2>&1 | grep -Eo '[0-9.]+'") ] >> 'versions'

    shell:
    meta = input.v1
    id = input.v2

    if (certificate) {
        if (certificate.toString().endsWith('.jwt')) {
            prefetch_args += " --perm ${certificate}"
        }
        else if (certificate.toString().endsWith('.ngc')) {
            prefetch_args += " --ngc ${certificate}"
        }
    }

    template 'retry_with_backoff.sh'
}
