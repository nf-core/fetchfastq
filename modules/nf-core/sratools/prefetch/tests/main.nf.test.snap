{
    "sratools/prefetch/stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "DRR000774.sra:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,83d1b23f5ff5b2ad1b96d17d7d7594ee"
                ],
                "sra": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "DRR000774.sra:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,83d1b23f5ff5b2ad1b96d17d7d7594ee"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-17T09:24:17.373992"
    },
    "sratools/prefetch with sralite": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "SRR1170046.sralite:md5,7acfce556ca0951aff49d780899c105b"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,83d1b23f5ff5b2ad1b96d17d7d7594ee"
                ],
                "sra": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "SRR1170046.sralite:md5,7acfce556ca0951aff49d780899c105b"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,83d1b23f5ff5b2ad1b96d17d7d7594ee"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-17T09:24:02.118978"
    },
    "sratools/prefetch": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "DRR000774.sra:md5,7647dba20c89c0e3d7ad13842f060eb0"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,83d1b23f5ff5b2ad1b96d17d7d7594ee"
                ],
                "sra": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "DRR000774.sra:md5,7647dba20c89c0e3d7ad13842f060eb0"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,83d1b23f5ff5b2ad1b96d17d7d7594ee"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-17T09:21:54.739086"
    }
}